backend-rust/src/indexer.rs

```diff
@@ -0,0 +1,1342 @@
+#![allow(unused_variables)] // TODO Remove before first release
+#![allow(dead_code)] // TODO Remove before first release
+
```

Comment:
I want to point out another thing with regards to these very large files. Since everything here is in the same module, everything is public to everything, so there is no encapsulation possible. Generally speaking, encapsulation is good and helps other good principles like separation of concerns and single responsibility.

Even if these principles are followed right now despite the single-module structure, the code is brittle to problems with these principles in the future. Separating the concerns would better protect against "code rot" or "spaghettification" as this code is maintained in the long term.

Splitting into multiple modules is therefore not just a question of style, but also of code cleanliness and maintainability.

Reply:
On a high level, this all sounds good, and I would not be against splitting up the file. I don't think the number of lines alone should be the argument for doing it, though, and it should be driven by an actual separation of concern.
I would prefer to have a big chunk of the implementation ready, instead of attempting to abstract ahead, then we can have actual concrete suggestions and avoid talking abstract principles.

Reply:
For some concrete examples, it could make sense to separate each query into a seperate module with its associated types, like the `BlockMetrics` and `BlockMetricsBuckets` types and the `block_metrics` method. Probably you could similarly split other queries out to modules. I think it would be most sensible/easiest to do this gradually (i.e. now, not as a "big bang" change later) as we build the new solution.

Reply:
Let's move this discussion offline instead